app/models/user.rb

```diff
@@ -1,9 +1,11 @@
 class User < ApplicationRecord
-  has_many :posts
-  has_many :likes
-  has_many :comments
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :likes, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, foreign_key: 'author_id', dependent: :destroy
 
+  validates :name, presence: true
+  validates :posts_counter, comparison: { greater_than_or_equal_to: 0 }
```

Comment:
- [ ] please add validation for post_counter being an integer

Reply:
Noted.